common/utils.js

```diff
@@ -1724,6 +1737,18 @@
             }
         }
 
+        function addCanonicalTag() {
+            var chaiseConfig = ConfigUtils.getConfigJSON();
+            if (chaiseConfig['includeCanonicalTag'] == true) {
+                var canonicalTag = document.createElement("link");
+                canonicalTag.setAttribute("rel", "canonical");
+                var canonicalURL = $window.location.origin + $window.location.pathname + UriUtils.stripSortAndQueryParams($window.location.hash);
```

Comment:
The window location might be using `?` for the hash portion of the url (remember the changes that we made to the UriUtils to make sure they call the same function that is taking care of this logic).
But why do you need to reconstruct the whole url piece by piece? Why are you not passing the whole `$window.location.href` to the `stripSortAndQueryParams` function that you've wrote?
Then that function should take care of the rest.

Reply:
I added strip query params after I had done the first part and didn't realize I could just call it on the whole url.